---
title: "Compiler Error CS0269 | Microsoft Docs"
ms.custom: ""
ms.date: "12/03/2016"
ms.prod: "visual-studio-dev14"
ms.reviewer: ""
ms.suite: ""
ms.technology: 
  - "devlang-csharp"
ms.tgt_pltfrm: ""
ms.topic: "error-reference"
f1_keywords: 
  - "CS0269"
dev_langs: 
  - "CSharp"
helpviewer_keywords: 
  - "CS0269"
ms.assetid: 7ef8374c-6f82-4096-bf4b-70080d4ddf88
caps.latest.revision: 14
caps.handback.revision: 14
author: "BillWagner"
ms.author: "wiwagn"
manager: "wpickett"
---
# Compiler Error CS0269
[!INCLUDE[vs2017banner](../../../csharp/includes/vs2017banner.md)]

Use para fora do parâmetro não atribuídos “parâmetro”  
  
 O compilador não conseguiu verificar se o parâmetro de saída foi atribuído um valor antes que esteve usado; o valor pode ser indefinido quando atribuído.  Certifique\-se de atribuir um valor aos parâmetros de `out` no método chamado antes de acessar o valor.  Se você precisar usar o valor da variável passado, use um parâmetro de `ref` em vez disso.  Para obter mais informações, consulte [Passando parâmetros](../../../csharp/programming-guide/classes-and-structs/passing-parameters.md).  
  
## Exemplo  
 O seguinte exemplo gera CS0269:  
  
```c#  
// CS0269.cs  
class C  
{  
    public static void F(out int i)  
    // One way to resolve the error is to use a ref parameter instead  
    // of an out parameter.  
    //public static void F(ref int i)  
    {  
        // The following line causes a compiler error because no value  
        // has been assigned to i.  
        int k = i;  // CS0269  
        i = 1;  
        // The error does not occur if the order of the two previous   
        // lines is reversed.  
    }  
  
    public static void Main()  
    {  
        int myInt = 1;  
        F(out myInt);  
        // If the declaration of method F is changed to require a ref  
        // parameter, ref must be specified in the call as well.  
        //F(ref myInt);  
    }  
}  
```  
  
## Exemplo  
 Isso também pode ocorrer se a inicialização de uma variável ocorre em um bloco try, que o compilador não pôde de verificação executa com êxito:  
  
```c#  
// CS0269b.cs  
class C  
{  
    public static void F(out int i)  
    {  
        try  
        {  
            // Assignment occurs, but compiler can't verify it  
            i = 1;  
        }  
        catch  
        {  
        }  
  
        int k = i;  // CS0269  
        i = 1;  
    }  
  
    public static void Main()  
    {  
        int myInt;  
        F(out myInt);  
    }  
}  
```